src/Components/Workspace/index.js

```diff
@@ -91,6 +101,15 @@ const WorkspaceView = props => {
     return () => {};
   }, [ id, props.firebase ]);
 
+  useEffect(() => {
+    window.onbeforeunload = (event) => {
+      event.preventDefault();
+      if (uploadTasks.size !== 0) {
+        event.returnValue = 'Your file has not finished uploading';
```

Comment:
This only results in a custom message in Safari; there aren't ways to add custom messages for Firefox and Chome.